lib/broadway_cloud_pub_sub/pull_client.ex

```diff
@@ -197,6 +201,11 @@ defmodule BroadwayCloudPubSub.PullClient do
       {:ok, %Response{status: 200, body: body}} ->
         {:ok, Jason.decode!(body)}
 
+      {:ok, %Response{status: status}}
+      when status in @retry_codes and retry_attempts < @max_retries ->
```

Comment:
This seems ok and will stop the noise for most people, however, it would be nice to be able to configure this.